src/Sylius/Behat/Context/Setup/OrderContext.php

```diff
@@ -767,49 +800,30 @@ public function theProductSInventoryHasBecameTrackedWithItems(ProductInterface $
         $this->objectManager->flush();
     }
 
-    /**
-     * @param string $transition
-     */
-    private function applyShipmentTransitionOnOrder(OrderInterface $order, $transition)
+    private function applyShipmentTransitionOnOrder(OrderInterface $order, string $transition): void
     {
         foreach ($order->getShipments() as $shipment) {
             $this->stateMachineFactory->get($shipment, ShipmentTransitions::GRAPH)->apply($transition);
         }
     }
 
-    /**
-     * @param string $transition
-     */
-    private function applyPaymentTransitionOnOrder(OrderInterface $order, $transition)
+    private function applyPaymentTransitionOnOrder(OrderInterface $order, string $transition): void
     {
         foreach ($order->getPayments() as $payment) {
             $this->stateMachineFactory->get($payment, PaymentTransitions::GRAPH)->apply($transition);
         }
     }
 
-    /**
-     * @param string $transition
-     */
-    private function applyTransitionOnOrderCheckout(OrderInterface $order, $transition)
+    private function applyTransitionOnOrderCheckout(OrderInterface $order, string $transition): void
```

Comment:
Looks like it needs `/** @throws SMException */` as well.

Reply:
We've already merged some `SyliusAbstractStateMachine` improvements
https://github.com/Sylius/Sylius/pull/15858/files#diff-18ea731fa30e78f757c08c3415360eb0824131d371b7184c2e45127044d33e2fR50
 and from now we introduced a new exception:
<img width="613" alt="image" src="https://github.com/Sylius/Sylius/assets/40125720/1b8cc8ba-b533-4b01-9ff1-d54c85c59f13">

Fun fact: static analysis no longer complains about unhandled exceptions, which I personally don't understand